src/utils/unique-string-array.js

```diff
@@ -38,6 +38,14 @@ export class UniqueStringArray {
     return index;
   }
 
+  changeString(index: IndexIntoStringTable, s: string) {
+    if (!(index in this._array)) {
+      throw new Error(`index ${index} not in UniqueStringArray`);
+    }
+    this._stringToIndex.set(s, index);
+    this._array[index] = s;
+  }
+
```

Comment:
I don't think we want to use this. The problem with this method is that the main predicate could now fail. This predicate is: there are only unique strings in this array, and 2 indexes are equals if and only if the 2 strings are equal.

See the comment in the upgrade function to see how we can handle this.